.github/workflows/deploy_release.yaml

```diff
@@ -51,8 +52,21 @@ jobs:
       - name: 'Check out repository'
         uses: actions/checkout@v4
         with:
-          ref: v${{ github.event.inputs.release-version }}
           token: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+          fetch-tags: true
+          fetch-depth: 0
+
+      - name: 'Get the branch associated with the PR'
+        env:
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+        run: |
+            echo "PR_HEAD=$(gh pr view ${{ inputs.release-pr-issue-number}} --json headRefName | jq -r .headRefName)" >> $GITHUB_ENV
+
+      - name: 'Get release tag from the branch of PR'
+        env:
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+        run: |
+          echo "RELEASE_VERSION=$(git name-rev --no-undefined --tags --name-only  origin/${PR_HEAD}^ | sed -e 's/^v//')" >> $GITHUB_ENV
 
```

Comment:
I want a check here that the head of main hasn't moved since the release was staged.  I haven't figured out the exactly `git` incantation.

Reply:
I've got no idea what the correct git interaction would be but I know we can enable checking that the branch is fully up to date with main as part of the branch protection rules.

Reply:
That sounds like a good idea.  I think I still want to preflight the promote.  We want to know that the PR will apply before we start the promoting the maven artefacts.

I have a simple idea for the preflight check.